airbyte-commons-protocol/src/main/java/io/airbyte/commons/protocol/migrations/AirbyteMessageMigration.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.commons.protocol.migrations;
+
+import io.airbyte.commons.version.AirbyteVersion;
+
+/**
+ * AirbyteProtocol message migration interface
+ *
+ * @param <Old> The Old AirbyteMessage type
+ * @param <New> The New AirbyteMessage type
+ */
+public interface AirbyteMessageMigration<Old, New> {
```

Comment:
Nit:  perhaps a better name for the `Old`/`New` objects would be `PreviousVersion` and `CurrentVersion` so that it's more clear what they represent.

Reply:
Updated